espnet2/train/preprocessor.py

```diff
@@ -1523,15 +1523,16 @@ def __init__(
         short_noise_thres: float = 0.5,
     ):
         super().__init__(train, rir_scp=rir_scp, rir_apply_prob=rir_apply_prob)
-        with open(spk2utt, "r") as f_s2u:
-            self.spk2utt = f_s2u.readlines()
+        if train:
+            with open(spk2utt, "r") as f_s2u:
+                self.spk2utt = f_s2u.readlines()
+            self._make_label_mapping()
+            self.nspk = len(self.spk2utt)
 
-        self.nspk = len(self.spk2utt)
         self.spk2label = None  # a dictionary that maps string speaker label to int
         self.sample_rate = sample_rate
         self.target_duration = int(target_duration * sample_rate)
         self.num_eval = num_eval
```

Comment:
I think you need to move the line `self.spk2label = None` to be above the `if train:` branch. Otherwise, `self.spk2label` will always be None.